### PR TITLE
Added arclight and nanolimbo types and Meowice flags

### LINKED ISSUE #1633: Add server type NanoLimbo
### Enhancement Type

A completely new feature

### Describe the enhancement

https://github.com/Nan1t/NanoLimbo

as Limbo server

### LINKED ISSUE #1953: Support for Arclight to manage Mod+Plugin in recent versions
### Enhancement Type

New server type

### Describe the enhancement

Like Magma, and better in higher MC version.

Arclight is almost the only type of Mod+Plugin bukkit in the latest version.

### PR DESCRIPTION
**Pull request by kingcavespider1**
Added Arclight should close #1953 

Added NanoLimbo closes #1633 

Was going to try to get new magma working too but the api doesn't seem to have an easy way to select a minecraft version without scaning thru the JSON

### CODE REVIEW

**Comment by kingcavespider1** [comment]
Build errors are because tests try to download modpacks and don't understand limbo type

nvm it was because custom jar file didn't have default loader type

**Comment by kingcavespider1** [comment]
Latest commits should close #3225

**Comment by kingcavespider1** [comment]
think that finally worked

**Comment by kingcavespider1** [comment]
java 21 takes forever to build

**Review by itzg — COMMENTED** [review]
Thanks for adding all these.

**Comment by itzg** [comment]
> java 21 takes forever to build

Wow that sure does. I'll need to investigate that some more. It's building for two architectures

https://github.com/itzg/docker-minecraft-server/blob/e70b9600d102848ba862c79fa1fe7d868fcc86dc/.github/workflows/verify-pr.yml#L29

but then so is java21-alpine.

**Comment by itzg** [comment]
Looking at other PR builds

https://github.com/itzg/docker-minecraft-server/pull/3484/checks

It might have been just a slow build instance.

**Comment by kingcavespider1** [comment]
Yeah seemed like most of the build time was just setting up ubuntu

File: scripts/start-deployNanoLimbo [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+. ${SCRIPTS:-/}start-utils
+isDebugging && set -x
+
+latestAsset=$(
+  curl -fsSL https://api.github.com/repos/Nan1t/NanoLimbo/releases/latest | \
```
[the remark this answers is not in the document]
Comment (kingcavespider1) [reply scripts/start-deployNanoLimbo:9]: I'll look thru Image helpers features more, I was mostly just looking into the modpack downloading part

**Review by itzg — DISMISSED** [review]
Thanks for switching over to the image helper usage. Any other changes you want before I merge? Got everything tested?

**Comment by kingcavespider1** [comment]
Can't test all edge cases but didn't run into any issues building or running it locally so should be ready for merge

In the future an option for getting a specific build from mc-image-helper github might be nice and should help clean up any of the scripts downloading from github. Mabye a regex for releases could be nice if there are any others like catserver.

**Comment by kingcavespider1** [comment]
Didn't realize this was still open before pushing that to master, adds Meowice flags and should close #3408 

on a positive note java 21's build time looked much better this time

**Review by itzg — COMMENTED** [review]
Bonus feature :)

Can you also add a section just below

https://github.com/itzg/docker-minecraft-server/blob/66d0605502b16be2af3b6416d544b2e4d03ac1b9/docs/configuration/jvm-options.md#L134-L133

**Comment by kingcavespider1** [comment]
> Bonus feature :)
> 
> Can you also add a section just below
> 
> https://github.com/itzg/docker-minecraft-server/blob/66d0605502b16be2af3b6416d544b2e4d03ac1b9/docs/configuration/jvm-options.md#L134-L133

Done

**Comment by kingcavespider1** [comment]
It seems to be only GraalVM 24 that gets the error when using those flags, I'll add a check for that

File: scripts/start-finalExec [hunk 2]
```diff
@@ -151,23 +151,47 @@ if isTrue "${ENABLE_JMX}"; then
   log "JMX is enabled. Make sure you have port forwarding for ${JMX_PORT}"
 fi
 
+if isTrue "${USE_MEOWICE_FLAGS}"; then
+  java_major_version=$(java -version 2>&1 | awk -F[\".] '/version/ {print ($2=="1" ? $3 : $2)}')
```
Comment (itzg) [inline scripts/start-finalExec:155]: Please use this subcommand instead

https://github.com/itzg/mc-image-helper#java-release

**Review by itzg — COMMENTED** [review]
Thank you for separating out the graalvm flags. 

After the requested change let me know when this is ready to merge.

**Comment by kingcavespider1** [comment]
Another useful feature in mc-image-helper that I missed, testing it rn

**Comment by itzg** [comment]
> Another useful feature in mc-image-helper that I missed, testing it rn

Very easy to miss since apparently the latest code isn't using it anymore. I think I originally added it for Forge compatibility checks with Java 8.

**Comment by kingcavespider1** [comment]
done, and yeah doesn't appear to be used currently anywhere. Was thinking about adding a check to set default flags by java version if unset, but probably deviates too far from current behavior. I'll test it in another branch to not clutter things here

**Review by itzg — APPROVED** [review]
